src/akkudoktoreos/prediction/elecpriceakkudoktor.py

```diff
@@ -184,80 +151,116 @@ def _update_data(self, force_update: Optional[bool] = False) -> None:
+                highest_orig_datetime = orig_datetime
+
+            price_wh = value.marketpriceEurocentPerKWh / (100 * 1000) + charges_wh
+
+            existing_record = next((r for r in self.records if r.date_time == orig_datetime), None)
+            if existing_record:
+                # Update existing record
+                existing_record.elecprice_marketprice_wh = price_wh
+            else:
+                self.insert(
+                    0,
+                    ElecPriceDataRecord(date_time=orig_datetime, elecprice_marketprice_wh=price_wh),
+                )
+
+        # Generate history array for prediction
+        history = np.array(
+            [
+                record.elecprice_marketprice_wh
+                for record in sorted(self.records, key=lambda r: r.date_time)
+                if record.elecprice_marketprice_wh is not None
+                and record.date_time
+                < highest_orig_datetime  # make sure we only real data for the prediction, so cant be newer then data from the api.
+            ]
+        )
+
+        amount_datasets = len(self.records)
+        assert highest_orig_datetime  # mypy fix
+
+        if amount_datasets > 800:
+            prediction = self._predict_ets(
+                history, seasonal_periods=168, prediction_hours=7 * 24
+            )  # todo: add config values for prediction_hours
+        elif amount_datasets > 168:
+            prediction = self._predict_ets(
+                history, seasonal_periods=24, prediction_hours=7 * 24
+            )  # todo: add config values for prediction_hours
+        elif (
+            amount_datasets > 0
+        ):  # TODO might be a problem if amount_datasets is really low and we do the _cap_outliers.
+            prediction = self._predict_median(history, prediction_hours=7 * 24)
+        else:
+            assert False, "No data available"
+
+        # write predictions into the records, update if exist.
+        for i, price in enumerate(prediction):
+            pred_datetime = highest_orig_datetime + to_duration(f"{i + 1} hours")
+            existing_record = next((r for r in self.records if r.date_time == pred_datetime), None)
+            if existing_record:
+                # Update existing record
+                existing_record.elecprice_marketprice_wh = price
+            else:
+                assert pred_datetime  # mypy fix, why do we need that we already made sure highest_orig_datetime is not None
+                self.insert(
+                    0,
+                    ElecPriceDataRecord(date_time=pred_datetime, elecprice_marketprice_wh=price),
+                )
+        history2 = np.array(  # TODO: remove return, only for debug, offset to see the difference
+            [
+                record.elecprice_marketprice_wh + 0.0002
+                for record in sorted(self.records, key=lambda r: r.date_time)
+                if record.elecprice_marketprice_wh is not None
+            ]
```

Comment:
Use `self.key_to_array(key="elecprice_marketprice_wh", fill_method="linear") + 0.0002` to get history2.